pkg/config/spec.go

```diff
@@ -323,7 +324,21 @@ func NewUpgradeSpec(cfg *Config) (*v1.UpgradeSpec, error) {
 		}
 	}
 
+	// Deep look to see if upgrade.recovery == true in the config
+	// if yes, we set the upgrade spec "Entry" to "recovery"
```

Comment:
This happened because I tried to unify `--recovery` and `--boot-entry` handling in the code. At the end of the day, they we do almost the same thing: we replace an efi file.
But I didn't want to unify the options too because we have `--recovery` everywhere in docs and people may even expect it to work if they are used to it. So I switch from a boolean to a value for UpgradeSpec.Entry as soon as possible, to use the same code as much as possible.

Reply:
good call